backend/core/src/csv/csv-builder.service.ts

```diff
@@ -1,130 +1,330 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
+import { HttpException, HttpStatus, Injectable } from "@nestjs/common"
+import dayjs from "dayjs"
 import { CsvEncoder } from "./csv-encoder.service"
-import { Injectable } from "@nestjs/common"
-import { CSVFormattingType } from "./types/csv-formatting-type-enum"
-import { FormattingMetadata } from "./types/formatting-metadata"
-import { FormattingMetadataArray } from "./types/formatting-metadata-array"
-import { FormattingMetadataAggregate } from "./types/formatting-metadata-aggregate"
-import { FormattingMetadataAggregateFactory } from "./types/formatting-metadata-aggregate-factory"
+import { ApplicationDto } from "../applications/dto/application.dto"
+import { capitalizeFirstLetter } from "../libs/stringLib"
+
+const HEADER_COUNT_REG = new RegExp(/\s\(\d\)/)
 
 @Injectable()
 export class CsvBuilder {
-  constructor(private readonly csvEncoder: CsvEncoder) {}
+  headers: { [key: string]: number }
+  data: { header: string; mappedField: string; val: any }[]
+  inProgress: boolean
+  excludeKeys: string[]
+  mappedFields: { [key: string]: string }
+  constructor(private readonly csvEncoder: CsvEncoder) {
+    this.headers = {}
+    this.data = []
+    this.inProgress = false
+    this.excludeKeys = [
+      " id",
```

Comment:
Why there are whitespaces in excludedKeys?

Reply:
because the keys consist of things like `householdMembers id`, but we do want `id`.